app/models/comment.php

```diff
@@ -47,6 +84,73 @@ public static function getAll($thread_id, $offset, $limit)
         return $comments;
     }
 
+    public static function getAllByUser(User $user)
+    {
+        $db     = DB::conn();
+        $rows   = $db->rows(
+            sprintf('SELECT * FROM %s WHERE user_id=? ORDER BY id DESC', self::TABLE_NAME),
+            array($user->id)
+        );
+
+        $comments = array();
+
+        foreach ($rows as $row) {
+            $comments[] = new self($row);
+        }
+
+        return $comments;
+    }
+
+
+    public static function get($id)
+    {
+        $db     = DB::conn();
+        $row    = $db->row(sprintf("SELECT * FROM %s WHERE id=?", self::TABLE_NAME), array($id));
+
+        return $row ? new self($row) : false;
+    }
+
+    public static function getOrFail($id)
+    {
+        $comment = self::get($id);
+
+        if ($comment) {
+            return $comment;
+        } else {
+            throw new RecordNotFoundException();
+        }
+    }
+
+    public static function getFirstInThread(Thread $thread)
+    {
+        $db     = DB::conn();
+        $row    = $db->row(sprintf('SELECT * FROM %s WHERE thread_id=? LIMIT 0, 1', self::TABLE_NAME), array($thread->id));
+
+        return $row ? new self($row) : false;
+    }
+
+    public static function getLastInThread(Thread $thread)
+    {
+        $db     = DB::conn();
+        $row    = $db->row(sprintf('SELECT * FROM %s WHERE thread_id=? ORDER BY id DESC LIMIT 0, 1', self::TABLE_NAME), array($thread->id));
+
+        return $row ? new self($row) : false;
+    }
```

Comment:
Suggestion po for getting the last comment in a thread, I think it is better to save the ID of the latest comment in thread table po, you would need to have a new column there ie. last_comment_id

With that po you do not need to get all the comments from a thread, sort it, only to get 1 row. It will only take time and slow the system. An extra work is needed though, since you would need to save the last comment ID in thread table every time a new comment is created. :bow: